system/CodeIgniter.php

```diff
@@ -969,8 +968,13 @@ protected function gatherOutput($cacheConfig = null, $returned = null)
 	 */
 	public function storePreviousURL($uri)
 	{
-		// Only valid for some requests
-		if (! $this->request instanceof IncomingRequest || $this->request->isCLI() || $this->request->isAJAX())
+		// Ignore CLI requests
+		if (method_exists($this->request, 'isCLI') && $this->request->isCLI())
```

Comment:
`is_cli()` is an available function provided in Common.php, so no need to check, just use the helper function. 

Unfortunately, none exists for is_ajax.